### ISSUE TITLE
MultiVI NaN in latent representation

### ISSUE THREAD

**Issue by ortholog** [scverse/scvi-tools#1422]
I encountered this error when I was running mvi.train():
```
mvi.train()
ValueError: Expected parameter loc (Tensor of shape (128, 11)) of distribution Normal(loc: torch.Size([128, 11]), scale: torch.Size([128, 11])) to satisfy the constraint Real(), but found invalid values:
tensor([[    nan,     nan,     nan,  ...,     nan,     nan,     nan],
        [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
        [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
        ...,
        [-0.6378, -0.6456,  0.6350,  ..., -0.2785, -0.6156, -0.0385],
        [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
        [-0.4364, -0.2716,  0.3634,  ...,  0.4258, -0.7137,  0.2691]],
       grad_fn=<AddmmBackward0>)

```
​I tried both datasets and encountered this error.Is this error because I used raw count?  Or other reasons?  
Please help me, I really want to know how to solve this problem.Thank you

**Comment by adamgayoso** [comment]
Can you describe more about your data? And the code you're using to run the model?

**Comment by ortholog** [comment]
> Can you describe more about your data? And the code you're using to run the model?
This is my scRNA-seq raw count:
![image](https://user-images.githubusercontent.com/86239688/157786552-88950db3-cafa-41bb-b614-96ea872d27af.png)
Then,I convert this count to anndata:
![image](https://user-images.githubusercontent.com/86239688/157786849-8dac0482-f978-42b3-a9f0-570e781f32c6.png)
And atac-seq data is processed in the same way.
![image](https://user-images.githubusercontent.com/86239688/157786971-c93d90cf-55b2-4194-adc6-4b092ea30e4e.png)
Then,scRNA-seq and ATAC-seq were integrated:
![image](https://user-images.githubusercontent.com/86239688/157787136-c25c18b9-54d0-4eef-a2f1-56ae8564800a.png)
Then follow the standard procedures provided on the website  :
![image](https://user-images.githubusercontent.com/86239688/157787270-6d90184b-352d-453b-8154-01949b337563.png)
![image](https://user-images.githubusercontent.com/86239688/157787314-2fb5a859-b16f-49ef-ac6e-fe9640a266db.png)

**Comment by adamgayoso** [comment]
If I'm understanding, you only have 778 cells, all of which you measured RNA and chromatin for?

**Comment by adamgayoso** [comment]
If this is the case, you would want to run multivi only on what you're calling`multi_adata` where the genes come first in `multi_adata.var`. However, it will be tough to learn a good model with only 778 cells.

**Comment by ortholog** [comment]
> If this is the case, you would want to run multivi only on what you're calling`multi_adata` where the genes come first in `multi_adata.var`. However, it will be tough to learn a good model with only 778 cells.

But I tried a dataset of more than 2,000 cells and encountered the same error

**Comment by adamgayoso** [comment]
The issue might be that your data is fully paired, yet you are running `organize_multiome_anndatas`. Can you try with just the fully paired data? Organized such that genes are before atac regions? Then give a batch key where every value is just "paired" or whatever you like.

**Comment by adamgayoso** [comment]
> But I tried a dataset of more than 2,000 cells and encountered the same error

2000 is also very few, can you add in any unpaired rna or atac datasets?

**Comment by ortholog** [comment]
My datasets are all paired, there are no unpaired datasets.How do you run MultiVI in this case?Could you elaborate on that? Thank you very much.

**Comment by ortholog** [comment]
> The issue might be that your data is fully paired, yet you are running `organize_multiome_anndatas`. Can you try with just the fully paired data? Organized such that genes are before atac regions? Then give a batch key where every value is just "paired" or whatever you like.

My datasets are all paired, there are no unpaired datasets.How do you run MultiVI in this case?Could you elaborate on that? Thank you very much.

**Comment by adamgayoso** [comment]
all you need to do is run multivi with anndata that has RNA features before ATAC features (don't run `organize_multiome_anndatas`). Then do 

`multi_adata.obs["modality"] = "paired"`

then 

```
model = MULTIVI.setup_anndata(multi_adata, batch_key="modality")
...
```
proceeding with the code you have

**Comment by adamgayoso** [comment]
I'm going to close this issue, we'd appreciate moving usage questions like this to:

https://discourse.scverse.org/

**Comment by ortholog** [comment]
> all you need to do is run multivi with anndata that has RNA features before ATAC features (don't run `organize_multiome_anndatas`). Then do
> 
> `multi_adata.obs["modality"] = "paired"`
> 
> then
> 
> ```
> model = MULTIVI.setup_anndata(multi_adata, batch_key="modality")
> ...
> ```
> 
> proceeding with the code you have
Thank you very much. It's very kind of you. I'll try it on at once

**Comment by ortholog** [comment]
> 

It worked! Thank you very much

**Comment by ortholog** [comment]
> all you need to do is run multivi with anndata that has RNA features before ATAC features (don't run `organize_multiome_anndatas`). Then do
> 
> `multi_adata.obs["modality"] = "paired"`
> 
> then
> 
> ```
> model = MULTIVI.setup_anndata(multi_adata, batch_key="modality")
> ...
> ```
> 
> proceeding with the code you have

It worked! Thank you very much

**Comment by ortholog** [comment]
Hello, at this website  https://docs.scvi-tools.org/en/stable/tutorials/notebooks/MultiVI_tutorial.html，scvi.settings.seed =420，If all my data are paired,Is the seed still set to 420? thank you


| |
guagua20210407
|
|
***@***.***
|
签名由网易邮箱大师定制
On 3/11/2022 10:32，Adam ***@***.***> wrote：

Closed #1422.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
